apps/prairielearn/src/pages/partials/studentAssessmentAsInstructor.ejs

```diff
@@ -0,0 +1,19 @@
+<% if (instance_role === 'Staff'
+       && authz_data.has_course_instance_permission_edit
```

Comment:
Is this necessary? I think if a user has staff access, regardless of course instance permissions, they should be able to do this. The course instance permissions would make sense if you're deleting a student assessment instance, but deleting your own should be fine.

Reply:
I don't know what `instance_role` does but I know that when I emulate another user or view their instance with that configuration, I don't see the alert. I don't actually check for "self" anywhere--should I?

Reply:
`instance_role` is the role of the user associated to the assessment instance has in the course instance. So `instance_role === 'Staff'` means the assessment instance is associated to a user with staff permission. This is not necessarily the same user currently authorized or authenticated, so a "self" option may be a good idea (i.e., check instance_user against the current user).

Reply:
That was going to be my comment: IMO we should only show this button if the instructor is the "owner" of an assessment, either directly or via group membership.

Reply:
You may want to take a look at group assessments as well, for those `instance_role` will not be defined.

Reply:
So, `resLocals.authz_result.authorized_edit` gets us halfway there, from what I can tell this indicates if the current user owns the assessment instance (or is in the group that does):

https://github.com/PrairieLearn/PrairieLearn/blob/9d3d0d0d7b4c514a116982e49662c7018dcc4317/apps/prairielearn/src/sprocs/authz_assessment_instance.sql#L110-L116

 I think `res.locals.authz_data.has_course_instance_permission_view` might be the other necessary piece here? That should essentially tell us if this user is an instructor (as opposed to a student, who should never be able to delete their own instances).

Does this sound correct? I would not at all be surprised if I was missing some subtlety here.

Reply:
I pushed this change in 2fc7714. Let's discuss.